src/main/java/org/primefaces/application/resource/csp/ContentSecurityPolicyFilter.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2009-2018 PrimeTek.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.application.resource.csp;
+
+import org.primefaces.application.resource.csp.scripts.ContentSecurityPolicyScripts;
+import org.primefaces.application.resource.csp.scripts.ContentSecurityPolicyScriptsResponseWriter;
+import org.primefaces.util.OnCommittedResponseWrapper;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * This servlet filter is responsible for setting appropriate Content-Security-Policy (CSP) response headers that are either whitelisted
+ * via {@link Constants.ContextParams#CONTENT_SECURITY_POLICY_HOST_WHITELIST} or determined automatically.
+ * Currently the <code>script-src</code> directive is supported only. It is planned to support other directives in the future as well
+ * that can then be specified via {@link Constants.ContextParams#CONTENT_SECURITY_POLICY_SUPPORTED_DIRECTIVES}.
+ * Per default browsers are instructed to report any CSP violations to the {@link ContentSecurityPolicyReportServlet} endpoint if not
+ * provided via {@link Constants.ContextParams#CONTENT_SECURITY_POLICY_REPORT_URI}.
+ */
+@WebFilter("/*")
```

Comment:
Does @WebFilter imply Servlet 3.0?  Would it work on Servlet 2.5 still?

Reply:
Both `@WebServlet` and `@WebFilter` annotations are defined in Servlet 3.0 specification, that's right. When looking at `pom.xml` PrimeFaces is already using Servlet 3.0 and JSF 2.3 so I am not sure if we have to support older versions at all?